### PR TITLE
ESP-DSP demo features for azure IoT dev board (BSP-296)

### PR DESCRIPTION
**Pull request by peter-marcisovsky**
# Change description
`esp32_azure_iot_kit` added I2C clk speed setter. Allowing faster display refresh rate. Supported setter for 100, 400 and 600 kHz. Also found out, `mag3110` has troubles communicating on fast I2C.

`ssd1306` added function to draw lines from provided XY coordinates.

`mag3110` added function to start sesnor without previous calibration. ESP-DSP demo is using Kalman filter, which does the calibration by itself.

`fbm320` added the highest OSR measurement

### CODE REVIEW

File: components/mag3110/include/mag3110.h [hunk 1]
```diff
@@ -101,6 +101,18 @@ void mag3110_delete(mag3110_handle_t sensor);
...
+ * @brief Start mag3110 sensor to measure, without previous sensor calibration
+ *
+ * @param sensor object handle of mag3110
+ * @param[in] data_rate Data rate and oversampling settings
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t mag3110_start_no_calibration(mag3110_handle_t sensor, const mag3110_data_rate_t data_rate);
```
Comment (tore-espressif) [inline components/mag3110/include/mag3110.h:114]:
```suggestion
esp_err_t mag3110_start_raw(mag3110_handle_t sensor, const mag3110_data_rate_t data_rate);
```
IMO, the description and name is not very clear. According to the datasheet 

> 0: Normal mode: data values are corrected by the user offset register values.
1: Raw mode: data values are not corrected by the user offset register values. 

So we should state here that the offset registers (that are set in `mag3110_calibrate` function) are ignore in raw mode

**Review by tore-espressif — CHANGES_REQUESTED** [review]
Thank you very much @peter-marcisovsky !
I left few comments, but nothing major

File: components/fbm320/fbm320.c [hunk 2]
```diff
@@ -174,7 +174,7 @@ esp_err_t fbm320_get_data(fbm320_handle_t sensor, const fbm320_measure_mode_t me
     if (ESP_OK != ret) {
         return ret;
     }
-    vTaskDelay(10 / portTICK_PERIOD_MS);
+    vTaskDelay(15 / portTICK_PERIOD_MS);
```
[the remark this answers is not in the document]
Comment (peter-marcisovsky) [reply components/fbm320/fbm320.c]: Changed to:
- `vTaskDelay(20 / portTICK_PERIOD_MS);` for `OSR8192`
- `vTaskDelay(10 / portTICK_PERIOD_MS);` for the rest of the measure modes

File: components/mag3110/include/mag3110.h (hunk 1 above)
Comment (peter-marcisovsky) [reply components/mag3110/include/mag3110.h]: changed to `mag3110_start_raw` as suggested

File: components/ssd1306/ssd1306.c [hunk 3]
```diff
@@ -242,6 +245,57 @@ void ssd1306_draw_bitmap(ssd1306_handle_t dev, uint8_t chXpos, uint8_t chYpos,
     }
 }
```
[the remark this answers is not in the document]
Comment (peter-marcisovsky) [reply components/ssd1306/ssd1306.c:247]: run pre-commit for formating

**Review by tore-espressif — APPROVED** [review]
Only one nitpick, rest LGTM!

**Comment by tore-espressif** [comment]
@peter-marcisovsky I'm sorry for the pre-commit troubles :/ We can merge after the formatting issue is resolved

**Pull request merged**